pkg/doc.go

```diff
@@ -0,0 +1,186 @@
+and are central to building Operators, Workload APIs, Configuration APIs, Autoscalers, and more.
+
+Client
+
+Client provides a Read / Write client for reading and writing objects to an apiserver.
+
+Cache
+
+Cache provides a Read client for reading objects from an apiserver that are stored in a local cache.
+Cache supports registering handlers to respond to events that update the cache.
+
+Manager
+
+Manager provides a mechanism for Starting components and provides shared Caches and Clients to the components.
+
+Controller
+
+Controller is a work queue that enqueues work in response to source.Source events (e.g. Pod Create, Update, Delete)
+and triggers reconcile.Reconcile functions when the work is dequeued.
+
+Unlike http handlers, Controllers DO NOT perform work directly in response to events, but instead enqueue
+reconcile.Requests so the work is performed eventually.
+
+* Controllers run reconcile.Reconcile functions against objects (provided as name / Namespace).
+
+* Controllers enqueue reconcile.Requests in response events provided by source.Sources.
+
+Reconcile
+
+reconcile.Reconcile is a function that may be called at anytime with the Name and Namespace of an
+object.  When called, it will ensure that the state of the system matches what is specified in the object at the
+time Reconcile is called.
+
+Example: Reconcile is run against a ReplicationController object.  The ReplicationController specifies 5 replicas.
+3 Pods exist in the system.  Reconcile creates 2 more Pods and sets their OwnerReference to point at the
+ReplicationController.
+
+* reconcile works on a single object type. - e.g. it will only reconcile ReplicaSets.
+
+* reconcile is triggered by a reconcile.Request containing the name / Namespace of an object to reconcile.
+
+* reconcile does not care about the event contents or event type triggering the reconcile.Request.
+- e.g. it doesn't matter whether a ReplicaSet was created or updated, reconcile will check that the correct
+Pods exist either way.
+
+* Users MUST implement Reconcile for Controllers.
+
+Source
+
+resource.Source provides a stream of events.  Events may be internal events from watching Kubernetes
+APIs (e.g. Pod Create, Update, Delete), or may be synthetic Generic events triggered by cron or WebHooks
+(e.g. through a Slackbot or GitHub callback).
+
+Example 1: source.Kind uses the Kubernetes API Watch endpoint for a GroupVersionKind to provide
+Create, Update, Delete events.
+
+Example 2: source.Channel reads Generic events from a channel fed by a WebHook called from a Slackbot.
+
+* source provides a stream of events for EventHandlers to handle.
+
+* source may provide either events from Watches (e.g. object Create, Update, Delete) or Generic triggered
+from another source (e.g. WebHook callback).
+
+* Users SHOULD use the provided Source implementations instead of implementing their own for nearly all cases.
+
+EventHandler
```

Comment:
We made changes to naming in `EventHandler` I think ?

Reply:
The interface stil EventHandler